Enterprise-CMCS/cmcs-eregulations
# [Tech Debt] Upgrade to Vite 5

**Description**

Vite 5 was [released on 16 November 2023](https://vitejs.dev/blog/announcing-vite5).  This major version release mostly introduces a changes under the hood, mainly performance upgrades via upgrading to Rollup4.  It also refactors some things to "future proof" this version of Vite.  [See here for more details](https://vitejs.dev/guide/migration).

We are currently on Vite ^4.5.

**This pull request changes:**

- Changes Vite version numbers in `package.json` files from `^4.5` to `^5.0`
- Handles CJS/CommonJS deprecation based on Migration Guide 
   - add `"type": "module"` in `package.json`
   - `import` instead of `require` in vite config
- Upgrade Vite's official Vue2 plugin to the [latest version that has Vite v5 support](https://github.com/vitejs/vite-plugin-vue2/blob/main/CHANGELOG.md)

**Steps to manually verify this change:**

1. Make sure experimental deployment succeeds and the site looks like you expect
2. Make sure all tests pass
3. Make sure it builds locally
4. Make sure vitest works as expected
